### PR TITLE
SDIT-1487 Separate create and update events to ensure failures captured correctly

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationService.kt [hunk 1]
```diff
@@ -28,7 +29,7 @@ class IncidentsSynchronisationService(
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  suspend fun synchroniseIncidentUpsert(event: IncidentsOffenderEvent) {
+  suspend fun synchroniseIncidentInsert(event: IncidentsOffenderEvent) {
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationService.kt:32]: Now have separate methods for insert and update

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationIntTest.kt [hunk 2]
```diff
@@ -176,56 +177,24 @@ class IncidentsSynchronisationIntTest : SqsIntegrationTestBase() {
...
+        internal fun `telemetry added to track the failure`() {
           incidentsNomisApi.stubGetIncident()
           incidentsMappingApi.stubGetIncident()
-          incidentsApi.stubIncidentUpsert()
 
           awsSqsIncidentsOffenderEventsClient.sendMessage(
             incidentsQueueOffenderEventsUrl,
             incidentEvent(eventType = "INCIDENT-INSERTED"),
           )
-        }
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationIntTest.kt:190]: Removed duplicate tests (not sure how that happened)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationIntTest.kt [hunk 3]
```diff
@@ -459,47 +402,6 @@ class IncidentsSynchronisationIntTest : SqsIntegrationTestBase() {
           }
         }
       }
-
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsSynchronisationIntTest.kt:462]: not needed for update

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**